usecases/event-driven-pipeline/README.md

```diff
@@ -0,0 +1,304 @@
+
+Run the command below to upload input data and pod templates. Once done, sparkjob-demo-bucket S3 bucket is created with two folders: input and scripts.
+```bash
+bash spark-scripts-data/upload-inputdata.sh
+```
+
+### Create a Step Functions state machine
+
+You need to make the following changes in sfn.yaml before apply. 
+
+* replace the value for roleARN with stepfunctions_role_arn 
+* replace the value for ExecutionRoleArn with emr_on_eks_role_arn
+* replace the value for VirtualClusterId with your virtual cluster id
+* optional: change sparkjob-demo-bucket with your bucket name 
+
+sfn.yaml
+```bash
+apiVersion: sfn.services.k8s.aws/v1alpha1
+kind: StateMachine
+metadata:
+  name: run-spark-job-ack
+spec:
+  name: run-spark-job-ack
+  roleARN: "arn:aws:iam::xxxxxxxxxxx:role/event-driven-pipeline-demo-sfn-execution-role"   # replace with your stepfunctions_role_arn
+  tags:
+  - key: owner
+    value: sfn-ack
+  definition: |
+      {
+      "Comment": "A description of my state machine",
+      "StartAt": "input-output-s3",
+      "States": {
+        "input-output-s3": {
+          "Type": "Task",
+          "Resource": "arn:aws:states:::emr-containers:startJobRun.sync",
+          "Parameters": {
+            "VirtualClusterId": "f0u3vt3y4q2r1ot11m7v809y6",  
+            "ExecutionRoleArn": "arn:aws:iam::xxxxxxxxxxx:role/event-driven-pipeline-demo-emr-eks-data-team-a",
+            "ReleaseLabel": "emr-6.7.0-latest",
+            "JobDriver": {
+              "SparkSubmitJobDriver": {
+                "EntryPoint": "s3://sparkjob-demo-bucket/scripts/pyspark-taxi-trip.py",
+                "EntryPointArguments": [
+                  "s3://sparkjob-demo-bucket/input/",
+                  "s3://sparkjob-demo-bucket/output/"
+                ],
+                "SparkSubmitParameters": "--conf spark.executor.instances=10"
+              }
+            },
+            "ConfigurationOverrides": {
+              "ApplicationConfiguration": [
+                {
+                 "Classification": "spark-defaults",
+                "Properties": {
+                  "spark.driver.cores":"1",
+                  "spark.executor.cores":"1",
+                  "spark.driver.memory": "10g",
+                  "spark.executor.memory": "10g",
+                  "spark.kubernetes.driver.podTemplateFile":"s3://sparkjob-demo-bucket/scripts/driver-pod-template.yaml",
+                  "spark.kubernetes.executor.podTemplateFile":"s3://sparkjob-demo-bucket/scripts/executor-pod-template.yaml",
+                  "spark.local.dir" : "/data1,/data2"
+                }
+              }
+              ]
+            }...
+```
+You can get your virtual cluster id from EMR console or use the command below.
+```bash
+kubectl get virtualcluster -o jsonpath={.items..status.id}
+# result:
+f0u3vt3y4q2r1ot11m7v809y6  # VirtualClusterId
```

Comment:
nit: Put this is another code block, so people can copy/paste without picking it up